FrankenPHP compatibility: SAPI check causes HTTP/2 panic during installation

## Summary

When running Maho on FrankenPHP (used in the official Docker images), the web installer crashes with a Go panic when installing with sample data. The root cause is that FrankenPHP's SAPI name is not recognized, causing a fallback to `flush()` which triggers an HTTP/2 panic.

## Error

```
panic: Header called after Handler finished

goroutine 99 [running, locked to thread]:
golang.org/x/net/http2.(*responseWriter).FlushError(0x3018720?)
    /root/go/pkg/mod/golang.org/x/net@v0.48.0/http2/server.go:2880 +0x10c
...
github.com/dunglas/frankenphp.go_sapi_flush(0x4000941e01?)
    /go/src/app/frankenphp.go:614 +0x90
```

## Root Cause

FrankenPHP implements `fastcgi_finish_request()` for compatibility, but reports its SAPI name as `frankenphp` instead of `fpm-fcgi`.

The current code in `app/code/core/Mage/Core/Model/App.php:349` checks:

```php
if (php_sapi_name() == 'fpm-fcgi' && function_exists('fastcgi_finish_request')) {
    fastcgi_finish_request();
} else {
    flush();
}
```

Since `php_sapi_name()` returns `frankenphp`, the condition fails and falls back to `flush()`, which causes the HTTP/2 panic in FrankenPHP/Caddy.

## Affected Files

1. `app/code/core/Mage/Core/Model/App.php:349`
2. `app/code/core/Mage/Install/controllers/WizardController.php:250-254`

## Proposed Fix

Update the SAPI check to include `frankenphp`:

```php
if (in_array(php_sapi_name(), ['fpm-fcgi', 'frankenphp'], true) && function_exists('fastcgi_finish_request')) {
    fastcgi_finish_request();
} else {
    flush();
}
```

Or simply remove the SAPI check and rely only on function existence:

```php
if (function_exists('fastcgi_finish_request')) {
    fastcgi_finish_request();
} else {
    flush();
}
```

## Environment

- Maho Docker images (`mahocommerce/maho:nightly`)
- FrankenPHP with PHP 8.5
- Occurs during web installer with sample data

--- a/app/code/core/Mage/Core/Model/App.php
+++ b/app/code/core/Mage/Core/Model/App.php
@@ -346,7 +346,7 @@ class Mage_Core_Model_App
         $this->getFrontController()->dispatch();
 
         // Finish the request explicitly, no output allowed beyond this point
-        if (php_sapi_name() == 'fpm-fcgi' && function_exists('fastcgi_finish_request')) {
+        if (in_array(php_sapi_name(), ['fpm-fcgi', 'frankenphp'], true) && function_exists('fastcgi_finish_request')) {
             fastcgi_finish_request();
         } else {
             flush();

